frontend/src/pages/Room/index.js

```diff
@@ -1,28 +1,56 @@
-import { useParams } from 'react-router-dom';
+import { useContext, useEffect } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
 
-import Button from "@mui/material/Button";
-import Grid from "@mui/material/Grid";
-import Typography from "@mui/material/Typography";
+import Button from '@mui/material/Button';
+import Grid from '@mui/material/Grid';
+import Typography from '@mui/material/Typography';
+import io from 'socket.io-client';
 
-import ChatBlock from "../../components/ui/chat/ChatBlock";
-import QuestionBox from "../../components/ui/question/QuestionBox";
+import AlertContext from '../../context/alert-context';
+import ChatBlock from '../../components/ui/chat/ChatBlock';
+import QuestionBox from '../../components/ui/question/QuestionBox';
 import CollabEditor from "../../components/ui/collaboration/CollabEditor";
-import styles from "./Room.module.css";
+import styles from './Room.module.css';
+
+import { URL_COMM_SVC } from '../../configs';
+
+const socket = io(URL_COMM_SVC);
 
 function RoomPage() {
-  // Todo: Use room id prop/state instead in the future
-  const { roomId } = useParams();
+  const alertCtx = useContext(AlertContext);
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const roomId = location.state?.room;
+
+  useEffect(() => {
+    if (!roomId) {
```

Comment:
Nice catch, I should do a similar logic for missing `difficulty` during matching also.

Reply:
Siuuuuuu

Reference: https://knowyourmeme.com/memes/cristiano-ronaldo-celebration-siuuuu